### PR TITLE
EZP-30845: Fixed importing XSLT stylesheet file with spaces in path

### PR DESCRIPTION
**Pull request by kmadejski**
| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | [EZP-30845](https://jira.ez.no/browse/EZP-30845)
| **Bug/Improvement**| yes
| **New feature**    | no
| **Target version** | `6.7`/`6.13`
| **BC breaks**      | no
| **Tests pass**     | yes
| **Doc needed**     | no

This PR fixes bug described in the JIRA ticket. Please refer to the ticket for more information. 

There is work on progress on the second PR for `ezplatform-richtext` package for v2.x.

**TODO**:
- [x] Implement feature / fix a bug.
- [x] Fix new code according to Coding Standards (`$ composer fix-cs`).
- [x] Ask for Code Review.

### CODE REVIEW

File: eZ/Publish/Core/FieldType/RichText/Converter/Xslt.php [hunk 1]
```diff
@@ -77,8 +77,8 @@ protected function getXSLTProcessor()
             $hrefAttr = $xslDoc->createAttribute('href');
 
             // Prevents showing XSLTProcessor::importStylesheet() warning on Windows file system
-            $hrefAttr->value = str_replace('\\', '/', $stylesheet);
-
+            // & allows loading stylesheets if project's directory name contains space(s)
+            $hrefAttr->value = str_replace(['\\', ' '], ['/', '%20'], $stylesheet);
```
Comment (alongosz) [inline eZ/Publish/Core/FieldType/RichText/Converter/Xslt.php]: Q: https://github.com/ezsystems/ezplatform-richtext/pull/63/files#r316763372

**Comment by andrerom** [comment]
Welcome to Windows 😈 

@alongosz Seems we are missing AppVeyor testing on other repos 😏

**Comment by kmadejski** [comment]
Well... it seems that we have to jump back to the previous "weird" approach @alongosz 🙁

**Comment by alongosz** [comment]
> @alongosz Seems we are missing AppVeyor testing on other repos, might be relevant for a fee of them

I'm not really happy with AppVeyor. It's very unstable, most of the times crashes and it's impossible to restart tests from the AppVeyor itself. Moreover free plan doesn't support jobs.

I tried to move that to Travis, but it seems that one .travis.yaml would need to support separate test setups for Linux and Windows, which right now is not possible.

Maybe something for @ezsystems/qa-team to look at in the future (just rolling the ball further away ;) )

> Well... it seems that we have to jump back to the previous "weird" approach @alongosz

Does it fail on AppVeyor only or also on an actual Windows install?

**Comment by kmadejski** [comment]
> Does it fail on AppVeyor only or also on an actual Windows install?

No idea, I have no access to Windows machine ATM.

**Comment by alongosz** [comment]
> > Does it fail on AppVeyor only or also on an actual Windows install?
> 
> No idea, I have no access to Windows machine ATM.

@mateuszbieniek please save us!

**Comment by andrerom** [comment]
> Well... it seems that we have to jump back to the previous "weird" approach @alongosz

I don't think we need to jump back to anything weird here. Either:
- just unit tests needs to be adapted to not hard code linux path logic*
- or something else

\* _integration tests passes, so I guess there is a good chance this is due to unit tests expectations, or?_

**Comment by kmadejski** [comment]
I double-checked it one time more and I think we have to jump back to the solution proposed in https://github.com/ezsystems/ezpublish-kernel/pull/2746/commits/44bb8fb530b169c438c8724e3be4503601f4486f. 

Please take a look at the following paths: 
1) Made by `realpath`:
`"/Users/kamil/www/ezplatform v1/vendor/ezsystems/ezpublish-kernel/eZ/Publish/Core/FieldType/RichText/Resources/stylesheets/docbook/xhtml5/edit/core.xsl"`
2) Made by https://github.com/ezsystems/ezpublish-kernel/pull/2746/commits/44bb8fb530b169c438c8724e3be4503601f4486f:
`"/Users/kamil/www/ezplatform%20v1/app/../vendor/ezsystems/ezpublish-kernel/eZ/Publish/Core/FieldType/RichText/Resources/stylesheets/docbook/xhtml5/edit/core.xsl"`

As you can notice, `realpath` keeps white characters which is a problem for both Mac and (I guess) Windows as well.

**Comment by alongosz** [comment]
Hmm, so it seems there's an internal problem with `libxml` (or whatever lib provides XSLTProcessor). Usually a pair of `realpath` and `file_get_contents` works very well, but here file is being opened by `xsl:import` which apparently does that differently.

@kmadejski before you go with the original `str_replace`, please try `rawurlencode` as it was suggested in some other PR related to that. If this is `xsl:import`, and not file opening by PHP, then it makes sense to apply url encoding there.

**Comment by kmadejski** [comment]
@alongosz I've tried with `rawurlencode` in https://github.com/ezsystems/ezpublish-kernel/pull/2746/commits/ea02c5f644482ab717f7fb89f974481ef64ca2e0. It works fine on my Mac, I'll double-check it locally with someone with Windows machine.

**Comment by kmadejski** [comment]
@alongosz I checked it on Windows and... I don't have good news. None of the proposed solutions work, including even `str_replace` approach. So far, `rawurlencode` seems to be the best choice, but I would love to hear your opinion about the approach which we want to keep 🙂

**Comment by gggeek** [comment]
@kmadejski what is the simplest code snippet I can use to test this? I can spend some time trying out different combinations on windows

**Comment by kmadejski** [comment]
@gggeek unfortunately, there is no code snippet for that. 
The easiest thing to do is to rename your eZ Platform root directory and add a space to it.

**Comment by adamwojs** [comment]
Closing PR as obsolete. eZ Platform 2.5 has reached EOM, so please reopen PR in ezsystems/ezplatform-kernel (bug fixes) or ibexa/core (features/improvements) if issue is still valid for v3.3 / v4.x and you are willing to work on it.

**Pull request closed without merging**